### PR TITLE
Add Requirement Level backward compatibility clarifications

### PR DESCRIPTION
**Pull request by lmolkova**
Added backward-compatibility  section to requirement levels to address https://github.com/open-telemetry/opentelemetry-specification/pull/2594#pullrequestreview-992483333

/cc @bogdandrutu

### CODE REVIEW

File: specification/common/attribute-requirement-level.md [hunk 1]
```diff
@@ -67,3 +70,31 @@ Here are several examples of expensive operations to be avoided by default:
 - DNS lookup to populate `net.peer.name` if only IP address is available to the instrumentation. Caching lookup results does not solve the issue for all possible cases and should be avoided by default too.
 - forcing `http.route` calculation before HTTP framework calculates it
 - reading response stream to find `http.response_content_length` when `Content-Length` header is not available
+
+## Changing requirement level
+
+Requirement level change is backward compatible if and only if the set of cases when previous version populates it, is a subset of cases when new version would populate it.
```
[the remark this answers is not in the document]
Comment (pyohannes) [reply specification/common/attribute-requirement-level.md:76]: I think it needs to be made clear whether backwards compatibility refers to the telemetry producer (instrumentation) or to the telemetry consumer (backend). It seems to me the changes in this PR refer to consumers.

Changing from `Recommended` to `Required` is a breaking change for producers, but not for consumers. Changing from `Required` to `Recommended` is a breaking change for consumers, but not for producers.

[the remark this answers is not in the document]
Comment (lmolkova) [reply specification/common/attribute-requirement-level.md:76]: I don't think telemetry stability refers to telemetry producers at all. When a new version of semantic conventions is released, producers don't have to update, at least for a long time. They don't need to support the previous version, they don't depend on the telemetry they produce. 
If they decide to update, they do it along with the schema version update. If they don't support the schema version, [they can't update at all](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/telemetry-stability.md)

Almost any semantic convention change would require some action from instrumentations, but non of it seems breaking as long as it's versioned,

[the remark this answers is not in the document]
Comment (lmolkova) [reply specification/common/attribute-requirement-level.md:76]: When producer did not populate schema version, almost any change will break it: 
- adding new required/recommended attribute (which is considered non-breaking)
- attribute rename, which is not really classified, but [allowed](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/schemas/file_format_v1.0.0.md#rename_attributes-transformation)

I understand the concern though, but there is nothing we can do for producers the don't version telemetry if we want to evolve conventions

[the remark this answers is not in the document]
Comment (lmolkova) [reply specification/common/attribute-requirement-level.md:78]: Created #2606 to discuss further.

File: specification/common/attribute-requirement-level.md [hunk 2]
```diff
@@ -67,3 +70,33 @@ Here are several examples of expensive operations to be avoided by default:
...
+_Note: Requirement level compatibility applies to telemetry consumers only. When semantic convention is updated from version `X` to version `Y` in a backward-compatible manner, consumers that only support version `X` would be fully compatible with telemetry version `Y`._
+
+Requirement level change is backward compatible if and only if the set of cases when previous version populates it, is a subset of cases when new version would populate it.
+
+- Requirement level MAY be increased from relaxed to more strict level if it does not add new conditions that limit when attribute is populated. Examples of such changes are:
+  - `Optional` to `Recommended`
+  - `Recommended` to `Required`
+  - `Conditionally Required` to `Required`
+  - `Optional` to `Conditionally Required` is allowed if the condition does not restrict attribute from being populated if explicitly enabled.
+  - `Recommended` to `Conditionally Required` is allowed if the condition allows to populate the attribute when it's not satisfied
```
Comment (jack-berg) [inline specification/common/attribute-requirement-level.md:85]:
```suggestion
  - `Recommended` to `Conditionally Required` is allowed if the condition allows the attribute to be populated when it's not satisfied
```

File: specification/common/attribute-requirement-level.md [hunk 3]
```diff
@@ -67,3 +70,33 @@ Here are several examples of expensive operations to be avoided by default:
...
+
```
Comment (jack-berg) [inline specification/common/attribute-requirement-level.md:86]:
```suggestion
```

File: specification/common/attribute-requirement-level.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jack-berg) [reply specification/common/attribute-requirement-level.md:78]: I'd consider removing this sentence. It seems to restate the sentence on line 76 with less clarity.

File: specification/common/attribute-requirement-level.md [hunk 4]
```diff
@@ -67,3 +70,33 @@ Here are several examples of expensive operations to be avoided by default:
 - DNS lookup to populate `net.peer.name` if only IP address is available to the instrumentation. Caching lookup results does not solve the issue for all possible cases and should be avoided by default too.
 - forcing `http.route` calculation before HTTP framework calculates it
 - reading response stream to find `http.response_content_length` when `Content-Length` header is not available
+
+## Changing requirement level
+
+_Note: Requirement level compatibility applies to telemetry consumers only. When semantic convention is updated from version `X` to version `Y` in a backward-compatible manner, consumers that only support version `X` would be fully compatible with telemetry version `Y`._
```
Comment (jack-berg) [inline specification/common/attribute-requirement-level.md:76]: These rules only apply to stable semantic conventions, correct? Might be worth noting that.

**Review by jack-berg — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request closed without merging**